src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -874,6 +866,35 @@ string ReserveName(string prefix)
             // when it's known the label will always be followed by a statement.
             void MarkLabel(string label, bool emitSemicolon = true) => writer.WriteLine($"{label}:{(emitSemicolon ? ";" : "")}");
 
+            // Emits a goto to jump to the specified label.  However, if the specified label is the top-level done label indicating
+            // that the entire match has failed, we instead emit our epilogue, uncapturing if necessary and returning out of Go.
+            void Goto(string doneLabel)
+            {
+                if (doneLabel == topLevelDoneLabel)
```

Comment:
I believe this (depending on the pattern) can happen multiple times right? Will that mean that the generated code have a lot of UncaptureUntil(0) and return calls?